frontend/src/lib/components/AnnotationsOverlay/annotationsOverlayLogic.ts

```diff
@@ -144,7 +145,13 @@ export const annotationsOverlayLogic = kea<annotationsOverlayLogicType>([
             (s) => [s.relevantAnnotations, s.intervalUnit, s.dateRange, s.pointsPerTick],
             (relevantAnnotations, intervalUnit, dateRange, pointsPerTick) => {
                 return groupBy(relevantAnnotations, (annotation) => {
-                    return determineAnnotationsDateGroup(annotation.date_marker, intervalUnit, dateRange, pointsPerTick)
+                    return determineAnnotationsDateGroup(
+                        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                        annotation.date_marker!,
```

Comment:
I think this'd be safer if `relevantAnnotations` returned a type like `DatedAnnotationType[]` that would have the relevant _guarantee_ of `date_marker` being present